src/util/options.rs

```diff
@@ -109,19 +107,22 @@ pub const DEFAULT_MIN_NURSERY: usize = 2 << LOG_BYTES_IN_MBYTE;
 #[cfg(target_pointer_width = "64")]
 pub const DEFAULT_MAX_NURSERY: usize = (1 << 20) << LOG_BYTES_IN_MBYTE;
 
-/// The default nursery space size.
-#[cfg(target_pointer_width = "32")]
-pub const NURSERY_SIZE: usize = 32 << LOG_BYTES_IN_MBYTE;
 /// The default min nursery size. This does not affect the actual space we create as nursery. It is
 /// only used in the GC trigger check.
 #[cfg(target_pointer_width = "32")]
 pub const DEFAULT_MIN_NURSERY: usize = 2 << LOG_BYTES_IN_MBYTE;
-const DEFAULT_MAX_NURSERY_32: usize = 32 << LOG_BYTES_IN_MBYTE;
+/// The default max nursery size for 32 bits.
+pub const DEFAULT_MAX_NURSERY_32: usize = 32 << LOG_BYTES_IN_MBYTE;
```

Comment:
These numbers were based off of JikesRVM and an older DaCapo benchmark suite. I'm not sure if they reflect good defaults for the larger workloads + compressed pointers would also use these defaults instead of the 64-bit one (which might explain the performance issues you saw).

Reply:
Ah wait -- this was a bit confusing: for `estimate_virtual_memory_in_bytes`, if `force_use_contiguous_space` is `false` (which it is for compressed pointers and 32-bit VMs), then 32 MB is used as the max nursery size when creating the space. But then in the above code in `GCTrigger` the `DEFAULT_MAX_NURSERY` is used which is much larger than 32 MB for 64-bit VMs with compressed pointers.